kong/resource_kong_plugin.go

```diff
@@ -80,23 +82,38 @@ func resourceKongPluginCreate(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	plugin, err := pluginClient.Create(pluginRequest)
-	if err != nil {
-		if config.upsertResources && strings.Contains(err.Error(), "unique constraint violation") {
-			dbPlugin, err := findPlugin(
-				pluginClient, pluginRequest.Name, pluginRequest.ConsumerId, pluginRequest.RouteId, pluginRequest.ServiceId,
-			)
-			if err != nil {
-				return err
-			}
+	var pluginID string
+	if err = resource.Retry(config.retryTimeout, func() *resource.RetryError {
+		log.Printf("creating plugin %s", pluginRequest.Name)
 
-			d.SetId(dbPlugin.Id)
-			return resourceKongPluginUpdate(d, meta)
+		plugin, err := pluginClient.Create(pluginRequest)
+		if err != nil {
+			if config.upsertResources && strings.Contains(err.Error(), "unique constraint violation") {
```

Comment:
I'd write a `isContraintViolationError` function that checks the second part but feel free to ignore.